packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanDetail/KeyValuesTable.js

```diff
@@ -32,25 +33,66 @@ function parseIfJson(value) {
 
 type KeyValuesTableProps = {
   data: { key: string, value: any }[],
+  linksGetter: ?({ key: string, value: any }[], number) => { url: string, text: string }[],
 };
 
 export default function KeyValuesTable(props: KeyValuesTableProps) {
-  const { data } = props;
+  const { data, linksGetter } = props;
   return (
     <div className="KeyValueTable u-simple-scrollbars">
       <table className="u-width-100">
         <tbody className="KeyValueTable--body">
           {data.map((row, i) => {
-            const jsonTable = (
-              // eslint-disable-next-line react/no-danger
-              <div dangerouslySetInnerHTML={{ __html: jsonMarkup(parseIfJson(row.value)) }} />
-            );
+            const markup = {
+              __html: jsonMarkup(parseIfJson(row.value)),
+            };
+            // eslint-disable-next-line react/no-danger
+            const jsonTable = <div className="ub-inline-block" dangerouslySetInnerHTML={markup} />;
+            const links = linksGetter ? linksGetter(data, i) : null;
+            let valueMarkup;
+            if (links && links.length === 1) {
+              valueMarkup = (
+                <div>
+                  <a href={links[0].url} title={links[0].text} target="_blank" rel="noopener noreferrer">
+                    {jsonTable} <Icon className="KeyValueTable--linkIcon" type="export" />
+                  </a>
+                </div>
+              );
+            } else if (links && links.length > 1) {
+              const menuItems = (
+                <Menu>
+                  {links.map((link, index) => {
+                    const { text, url } = link;
+                    return (
+                      // `index` is necessary in the key because url can repeat
+                      // eslint-disable-next-line react/no-array-index-key
+                      <Menu.Item key={`${url}-${index}`}>
+                        <a href={url} target="_blank" rel="noopener noreferrer">
+                          {text}
+                        </a>
+                      </Menu.Item>
+                    );
+                  })}
+                </Menu>
+              );
+              valueMarkup = (
+                <div>
+                  <Dropdown overlay={menuItems} placement="bottomRight" trigger={['click']}>
+                    <a>
+                      {jsonTable} <Icon className="KeyValueTable--linkIcon" type="profile" />
+                    </a>
+                  </Dropdown>
+                </div>
+              );
+            } else {
+              valueMarkup = jsonTable;
+            }
             return (
               // `i` is necessary in the key because row.key can repeat
               // eslint-disable-next-line react/no-array-index-key
               <tr key={`${row.key}-${i}`}>
                 <td className="KeyValueTable--keyColumn">{row.key}</td>
-                <td>{jsonTable}</td>
+                <td>{valueMarkup}</td>
```

Comment:
Seems like the code above is sufficiently intricate to warrant being broken down into sub-components at the top of the file, similar to [`CustomNavDropdown`](https://github.com/jaegertracing/jaeger-ui/blob/e31801d1a283cc40b360cfda01e4e061a124c0d8/packages/jaeger-ui/src/components/App/TopNav.js#L45). Something like `LinkValue` and `LinkValueList` where `LinkValueList` would compose `LinkValue`? 

```jsx
// Individual link
<LinkValue
  href={href}
  title={title}
  content={jsonMarkup}
/>

// List of links
<LinkValueList
  links={links}
  content={jsonMarkup}
/>
```

What do you think?

Also, I think it would make sense for the link to always show the icon?

Reply:
I have taken into account this suggestion and updated the PR.
Note that the `Dropdown` component in its `overlay` property seems to work differently if there is a wrapper component such as `LinkValueList` before the `Menu` component, so I don't call `LinkValueList` as a component (`<LinkValueList ... />`) but directly as a function (`{linkValueList(...)}`).

Reply:
Looks great, thank you.